components/org.wso2.carbon.identity.context.rewrite.valve/src/main/java/org/wso2/carbon/identity/context/rewrite/valve/TenantContextRewriteValve.java

```diff
@@ -81,6 +81,10 @@ public void invoke(Request request, Response response) throws IOException, Servl
             }
         }
 
+        if (isTenantQualifiedModeEnabled()) {
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            IdentityUtil.threadLocalProperties.get().put(TENANT_NAME_FROM_CONTEXT, tenantDomain);
+        }
```

Comment:
This would add tenant to context in unintended flows right.
If a URI context is intended to have a tenant in path only we should set this context.

Reply:
This would only add the tenant domain to thread local. Reading and appending the tenant domain to urls is done via the ServiceURLBuilder.

Reply:
Well my concern is so far we have been setting this thread local property only when, a url context that has been configured as a context to re-write and had /t/ in path.

This thread local was not there at any other cases right.
Now irrespective of the fact that the context is configured if its tenant url mode you set it.

So it could be set in a context like not intended to be re-written.
There won't be a breaking flow, but the information will be available in the context unnecessarily which might be used later by some other party at development